packages/core/src/Keymap.ts

```diff
@@ -0,0 +1,101 @@
+import { CommandIdentifier, CommandArgs } from './Dispatcher';
+
+interface ShortcutPattern {
+    key: string;
+    modifiers: Set<string>;
+}
+// todo: should I place this in file Command?
+export interface CommandCall {
+    commandId: string;
+    commandArgs?: CommandArgs;
+}
+
+interface ShortcutItem extends CommandCall {
+    pattern: ShortcutPattern;
+    priority: number;
+}
+
+// todo: questions
+// - who preventDefault / stopPropagation / stopImmediatePropagation?
+// - what about when a pattern is already present in the system?
+//   - if the check (static or dynamic) is different?
+export class Keymap {
+    _shortcutItems: ShortcutItem[] = [];
+
+    //--------------------------------------------------------------------------
+    // public
+    //--------------------------------------------------------------------------
+
+    addShortuct(
+        patternString: string,
+        commandId: CommandIdentifier,
+        commandArgs?: CommandArgs,
+        priority = 0,
+    ): void {
+        this._shortcutItems.push({
+            pattern: this._getPatternFromString(patternString),
+            commandId,
+            commandArgs,
+            priority,
+        });
+    }
+
+    getCommandCalls(keyEvent: KeyboardEvent): CommandCall[] {
```

Comment:
I would have called this function `match` but I'm 99.99% sure you're going to disagree.